earth2studio/models/dx/cyclone_tracking.py

```diff
@@ -0,0 +1,1063 @@
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+        return OrderedDict(
+            {
+                "batch": np.empty(0),
+                "variable": np.array(VARIABLES_TCV),
+                "lat": np.linspace(90, -90, 721, endpoint=True),
+                "lon": np.linspace(0, 360, 1440, endpoint=False),
+            }
+        )
+
+    @batch_coords()
+    def output_coords(self, input_coords: CoordSystem) -> CoordSystem:
+        """Output coordinate system of diagnostic model
+
+        Parameters
+        ----------
+        input_coords : CoordSystem
+            Input coordinate system to transform into output_coords
+            by default None, will use self.input_coords.
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+        target_input_coords = self.input_coords()
+        handshake_dim(input_coords, "lon", 3)
+        handshake_dim(input_coords, "lat", 2)
+        handshake_dim(input_coords, "variable", 1)
+        handshake_coords(input_coords, target_input_coords, "lon")
+        handshake_coords(input_coords, target_input_coords, "lat")
+        handshake_coords(input_coords, target_input_coords, "variable")
+
+        output_coords = input_coords.copy()
+        output_coords.pop("lat")
+        output_coords.pop("lon")
+        output_coords["variable"] = np.array(OUT_VARIABLES)
+        output_coords["point"] = np.arange(0)
+
+        return output_coords
+
+    @classmethod
+    def load_model(cls, package: Package) -> DiagnosticModel:
+        """Load diagnostic from package"""
+
+        return cls()
+
+    @torch.inference_mode()
+    @batch_func()
+    def __call__(
+        self,
+        x: torch.Tensor,
+        coords: CoordSystem,
+    ) -> tuple[torch.Tensor, CoordSystem]:
+        """Forward pass of diagnostic"""
+
+        output_coords = self.output_coords(coords)
+
+        lat = torch.as_tensor(self.input_coords()["lat"], device=x.device)
+        lon = torch.as_tensor(self.input_coords()["lon"], device=x.device)
+
+        def get_variable(x: torch.Tensor, var: str) -> torch.Tensor:
+            index = VARIABLES_TCV.index(var)
+            return x[index]
+
+        outs = []
+        for i in range(x.shape[0]):
+
+            # Get wind components at 850 hPa
+            u850 = get_variable(x[i], "u850")
+            v850 = get_variable(x[i], "v850")
+            u10m = get_variable(x[i], "u10m")
+            v10m = get_variable(x[i], "v10m")
+            msl = get_variable(x[i], "msl")
+
+            # Calculate vorticity at 850 hPa
+            vort850 = vorticity(u850, v850)
+            vort850[361:] *= -1
+
+            # Calculate wind speed at 10m height
+            w10m = torch.sqrt(torch.square(u10m) + torch.square(v10m))
+
+            # identify position of tropical storm centers
+            centers = find_centers_wuduan(lat, lon, vort850, w10m, msl)
```

Comment:
Is this function only expected to be used in this model?

If so then perhaps some of these should be moved into the class, even as a class function